packages/fluentui/react-northstar/src/utils/positioner/Popper.tsx

```diff
@@ -292,6 +293,35 @@ export const Popper: React.FunctionComponent<PopperProps> = props => {
           fn: handleUpdate,
         },
         popperInitialPositionFix.modifier,
+
+        autoSize && {
+          // Similar code as popper-maxsize-modifier: https://github.com/atomiks/popper.js/blob/master/src/modifiers/maxSize.js
+          // popper-maxsize-modifier only calculates the max sizes. This modifier applies the max sizes when overflow is detected
+          name: 'applyMaxSize',
+          enabled: true,
+          phase: 'beforeWrite' as PopperJs.ModifierPhases,
+          requiresIfExists: ['offset', 'preventOverflow', 'flip'],
+          options: {
+            altBoundary: true,
+            boundary: getBoundary(contentRef.current, overflowBoundary),
+          },
+          fn({ state, options }: PopperJs.ModifierArguments<{}>) {
+            const overflow = PopperJs.detectOverflow(state, options);
+            const { x, y } = state.modifiersData.preventOverflow || { x: 0, y: 0 };
+            const { width, height } = state.rects.popper;
+            const [basePlacement] = state.placement.split('-');
+
+            const widthProp = basePlacement === 'left' ? 'left' : 'right';
+            const heightProp = basePlacement === 'top' ? 'top' : 'bottom';
```

Comment:
```suggestion
            const widthProp: keyof PopperJs.SideObject = basePlacement === 'left' ? 'left' : 'right';
            const heightProp: keyof PopperJs.SideObject = basePlacement === 'top' ? 'top' : 'bottom';
```

nit, it seems that Popper defines a type for this. Can we try to reuse it?

https://github.com/popperjs/popper-core/blob/509875b0392c036739f4fe85fa580e2f265c9def/src/utils/detectOverflow.js#L36